pandas/io/sas/sas7bdat.py

```diff
@@ -687,6 +687,26 @@ def _read_next_page(self):
 
         return False
 
+    @staticmethod
+    def _convert_datetimes(sas_datetimes, unit):
+        """Converts to np.datetime64 if possible, otherwise to dateime.datetime
+        (n.b. generally better support in pandas for datetime than date).
+        SAS float64 lacks precision for more than ms resolution so the fit
+        to datetime.datetime is ok
+        """
+        try:
+            return pd.to_datetime(sas_datetimes, unit=unit, origin="1960-01-01")
```

Comment:
we *already* have an ``errors`` keyword can you use this?

Reply:
Hi,

Not quite sure what you mean - are you thinking that the call to to_datetime (line 698 above) should have errors='ignore' and then use the apply to pick up any items that are still float64 (from SAS) and convert just those items to datetime.datetime?

I went for the try/except as the majority of SAS datasets probably only have dates within the bounds of Timestamp, and we then avoid the additional cost of the apply to check for remaining float64 values to convert.

On a slight aside, I notice also that the docstring for pd.to_datetime states
"In case when it is not possible to return designated types (e.g. when
        any element of input is before Timestamp.min or after Timestamp.max)
        return will have datetime.datetime type (or corresponding
        array/Series)."
This doesn't seem to be true, e.g.
import pandas as pd
import datetime

df = pd.DataFrame([{'max': '9999-12-31'}])
df['conv'] = pd.to_datetime(df['max'])

print(df)

raises OutOfBoundsDatetime rather than converting to a datetime.datetime for 9999-12-31

Reply:
Hi @jreback, I've merged master - could you take a look? Thanks.